src/Qowaiv/SvoBehavior.cs

```diff
@@ -0,0 +1,235 @@
+        => sourceType == typeof(string);
+
+    /// <inheritdoc />
+    [Pure]
+    public override bool CanConvertTo(ITypeDescriptorContext? context, Type? destinationType)
+        => destinationType == typeof(string);
+
+    /// <inheritdoc />
+    [Pure]
+    public override object? ConvertFrom(ITypeDescriptorContext? context, CultureInfo? culture, object value)
+        => TryParse(value?.ToString(), culture, out var parsed)
+        ? parsed
+        : throw InvalidFormat(value?.ToString(), culture);
+
+    /// <inheritdoc />
+    [Pure]
+    public override object? ConvertTo(ITypeDescriptorContext? context, CultureInfo? culture, object? value, Type destinationType)
+        => ToString(value?.ToString(), default, culture);
+
+    /// <summary>Converts the <see cref="string"/> to a 
+    /// A return value indicates whether the conversion succeeded.
+    /// </summary>
+    /// <param name="str">
+    /// A string containing the Single Value Object to convert.
+    /// </param>
+    /// <param name="formatProvider">
+    /// The specified format provider.
+    /// </param>
+    /// <param name="validated">
+    /// The result of the parsing.
+    /// </param>
+    /// <returns>
+    /// True if the string was converted successfully, otherwise false.
+    /// </returns>
+    [Pure]
+    internal bool TryParse(string? str, IFormatProvider? formatProvider, out string? validated)
+    {
+        var normalized = NormalizeInput(str, formatProvider);
+        if (string.IsNullOrWhiteSpace(normalized))
+        {
+            validated = null;
+            return true;
+        }
+        else if (IsUnknown(normalized, formatProvider))
+        {
+            validated = unknown;
+            return true;
+        }
+        else if (WithinSize(normalized)
+            && IsMatch(normalized)
+            && IsValid(normalized, formatProvider, out validated))
+        {
+            return true;
+        }
+        else
+        {
+            validated = null;
+            return false;
+        }
+
+        bool WithinSize(string str)
+            => str.Length >= MinLength 
+            && str.Length <= MaxLength;
+
+        bool IsMatch(string str)
+            => Pattern is null
+            || Pattern.IsMatch(str);
+    }
+
+    /// <summary>Returns a formatted <see cref="string" /> that represents the Single Value Object.</summary>
+    /// <param name="str">
+    /// The string representing the Single Value Object.
+    /// </param>
+    /// <param name="format">
+    /// The format that this describes the formatting.
+    /// </param>
+    /// <param name="formatProvider">
+    /// The format provider.
+    /// </param>
+    [Pure]
+    internal string ToString(string? str, string? format, IFormatProvider? formatProvider)
+    {
+        if (StringFormatter.TryApplyCustomFormatter(format, str!, formatProvider, out string formatted))
+        {
+            return formatted;
+        }
+        else if (str is null)
```

Comment:
You don't want to do this check first? Or is this in case the custom formatter has specific null logic?

Reply:
Yes (some sort of). if an `ICustomFormatter` is provided, it is up to that formatter completely. It might contain logging (I would not recommand doing so), or preferring to return `null`, whatever.